lib/Cake/Test/Case/Cache/CacheTest.php

```diff
@@ -538,4 +538,64 @@ public function testAdd() {
 		$result = Cache::add('test_add_key', 'test data 2', 'default');
 		$this->assertFalse($result);
 	}
+
+/**
+ * Test engine method.
+ *
+ *  Success, default engine.
+ *
+ * @return void
+ */
+	public function testEngineSuccess() {
+		$actual = Cache::engine();
+		$this->assertIsA($actual, 'CacheEngine');
+
+		$actual = Cache::engine('default');
+		$this->assertIsA($actual, 'CacheEngine');
+	}
+
+/**
+ * Test engine method.
+ *
+ *  Success, memcached engine.
+ *
+ * @return void
+ */
+	public function testEngineSuccessMemcached() {
+		$this->skipIf(!class_exists('Memcached'), 'Memcached is not installed or configured properly.');
+
+		// @codingStandardsIgnoreStart
+		$socket = @fsockopen('127.0.0.1', 11211, $errno, $errstr, 1);
+		// @codingStandardsIgnoreEnd
+		$this->skipIf(!$socket, 'Memcached is not running.');
+		fclose($socket);
```

Comment:
I only asked because it adds another dependency for this test which makes it run in fewer environments.